worlds/wl4/rules.py

```diff
@@ -0,0 +1,269 @@
+from __future__ import annotations
+
+import functools
+from typing import Callable, Mapping, NamedTuple, Optional, Sequence, Tuple, Union, TYPE_CHECKING
+
+from BaseClasses import CollectionState
+from Options import Option
+
+from . import items, locations, options
+from .types import ItemType, Passage
+from .options import Goal
+
+if TYPE_CHECKING:
+    from . import WL4World
+
+
+RequiredItem = Union[str, Tuple[str, int]]
+
+
+helpers: Mapping[str, RequiredItem] = {
+    'Ground Pound':        'Progressive Ground Pound',
+    'Super Ground Pound': ('Progressive Ground Pound', 2),
+    'Grab':                'Progressive Grab',
+    'Heavy Grab':         ('Progressive Grab', 2)
+}
+
+
+def resolve_helper(item_name: RequiredItem):
+    requirement = helpers.get(item_name, item_name)
+    if isinstance(requirement, str):
+        return (requirement, 1)
+    else:
+        return requirement
```

Comment:
So because this accepts tuples of ints, you're calling `helpers.get` with a tuple of ints which isn't exactly supported here (it works, but it's kind of taking advantage of this _not_ causing an error). You could do this instead:
```suggestion
def resolve_helper(item_name: RequiredItem):
    if not isinstance(item_name, str):
        return item_name
    item_name = helpers.get(item_name, item_name)
    if not isinstance(item_name, str):
        return item_name
    return item_name, 1
```

Another option that I think is quite a bit cleaner is changing `helpers` to a `Mapping[str, Tuple[str, int]]` so you'd change `'Progressive Ground Pound'` to `('Progressive Ground Pound', 1)` and then just having this instead:
```suggestion
def resolve_helper(item_name: RequiredItem):
    if not isinstance(item_name, str):
        return item_name
    return helpers.get(item_name, (item_name, 1))
```